client/api_sp.go

```diff
@@ -0,0 +1,203 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"strings"
+	"time"
+
+	"cosmossdk.io/math"
+	"github.com/bnb-chain/greenfield-go-sdk/pkg/utils"
+	gnfdSdkTypes "github.com/bnb-chain/greenfield/sdk/types"
+	spTypes "github.com/bnb-chain/greenfield/x/sp/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	govTypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+)
+
+type SP interface {
+	// ListSP return the storage provider info on chain
+	// isInService indicates if only display the sp with STATUS_IN_SERVICE status
+	ListSP(ctx context.Context, isInService bool) ([]spTypes.StorageProvider, error)
+	// GetSPInfo return the sp info the sp chain address
+	GetSPInfo(ctx context.Context, SPAddr sdk.AccAddress) (*spTypes.StorageProvider, error)
```

Comment:
It is more convenient to pass in the string address of hex for users.

Reply:
updated